component/sidetree/client.go

```diff
@@ -0,0 +1,638 @@
+	}
+
+	if recoverDIDOpts.NextUpdatePublicKey == nil {
+		return fmt.Errorf("next update public key is required")
+	}
+
+	if recoverDIDOpts.SigningKey == nil {
+		return fmt.Errorf("signing key is required")
+	}
+
+	if len(recoverDIDOpts.SidetreeEndpoints) == 0 {
+		return fmt.Errorf("sidetree endpoints is required")
+	}
+
+	return nil
+}
+
+func validateDeactivateReq(deactivateDIDOpts *deactivate.Opts) error {
+	if deactivateDIDOpts.SigningKey == nil {
+		return fmt.Errorf("signing key is required")
+	}
+
+	if len(deactivateDIDOpts.SidetreeEndpoints) == 0 {
+		return fmt.Errorf("sidetree endpoints is required")
+	}
+
+	return nil
+}
+
+// buildCreateRequest request builder for sidetree public DID creation.
+func buildCreateRequest(multiHashAlgorithm uint, createDIDOpts *create.Opts) ([]byte, error) {
+	didDoc := &doc.Doc{
+		PublicKey: createDIDOpts.PublicKeys,
+		Service:   createDIDOpts.Services,
+	}
+
+	docBytes, err := didDoc.JSONBytes()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get document bytes : %s", err)
+	}
+
+	recoveryKey, err := pubkey.GetPublicKeyJWK(createDIDOpts.RecoveryPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get recovery key : %s", err)
+	}
+
+	updateKey, err := pubkey.GetPublicKeyJWK(createDIDOpts.UpdatePublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get update key : %s", err)
+	}
+
+	recoveryCommitment, err := commitment.GetCommitment(recoveryKey, multiHashAlgorithm)
+	if err != nil {
+		return nil, err
+	}
+
+	updateCommitment, err := commitment.GetCommitment(updateKey, multiHashAlgorithm)
+	if err != nil {
+		return nil, err
+	}
+
+	createRequestInfo := &client.CreateRequestInfo{
+		OpaqueDocument:     string(docBytes),
+		RecoveryCommitment: recoveryCommitment,
+		UpdateCommitment:   updateCommitment,
+		MultihashCode:      multiHashAlgorithm,
+	}
+
+	req, err := client.NewCreateRequest(createRequestInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create sidetree request: %w", err)
+	}
+
+	return req, nil
+}
+
+// buildUpdateRequest request builder for sidetree public DID update.
+func (c *Client) buildUpdateRequest(did string, multiHashAlgorithm uint,
+	updateDIDOpts *update.Opts) ([]byte, error) {
+	nextUpdateKey, err := pubkey.GetPublicKeyJWK(updateDIDOpts.NextUpdatePublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get next update key : %s", err)
+	}
+
+	nextUpdateCommitment, err := commitment.GetCommitment(nextUpdateKey, multiHashAlgorithm)
+	if err != nil {
+		return nil, err
+	}
+
+	signer, updateKey, err := getSigner(updateDIDOpts.SigningKey, updateDIDOpts.SigningKeyID)
+	if err != nil {
+		return nil, err
+	}
+
+	patches, err := createUpdatePatches(updateDIDOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	didSuffix, err := getUniqueSuffix(did)
+	if err != nil {
+		return nil, err
+	}
+
+	revealValue := updateDIDOpts.RevealValue
+
+	// TODO: client should be managing reveal value, this defaulting here is just temporary convenience
```

Comment:
Please create ticket to remove this code and force the clients of this sidetree client to manage reveal value. It is referenced in update, deactivate and recover.

Reply:
@sandrask I force clients of this sidetree client to manage reveal value to manage reveal value